ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/VertxPeerDiscoveryAgent.java

```diff
@@ -58,21 +61,33 @@ public VertxPeerDiscoveryAgent(
       final Vertx vertx,
       final KeyPair keyPair,
       final DiscoveryConfiguration config,
-      final PeerRequirement peerRequirement,
       final PeerBlacklist peerBlacklist,
       final Optional<NodeLocalConfigPermissioningController> nodeWhitelistController,
       final Optional<NodePermissioningController> nodePermissioningController,
       final MetricsSystem metricsSystem) {
     super(
         keyPair,
         config,
-        peerRequirement,
         peerBlacklist,
         nodeWhitelistController,
         nodePermissioningController,
         metricsSystem);
     checkArgument(vertx != null, "vertx instance cannot be null");
     this.vertx = vertx;
+
+    metricsSystem.createIntegerGauge(
```

Comment:
Just to clarify, is the VertxPeerDiscoveryAgent created once and only once per server?  If not the `createIntegerGauge` call will result in errors on the second and subsequent calls.  Not sure there is any way to hedge this.

Reply:
Yes, currently only created once.